src/main/java/teammates/logic/core/EmailGenerator.java

```diff
@@ -171,6 +171,22 @@ private EmailWrapper generateFeedbackSessionEmailBaseForInstructorReminders(
     }
     
     /**
+     * Generates the feedback session closed emails for the given {@code session}.
+     */
+    public List<EmailWrapper> generateFeedbackSessionClosedEmails(FeedbackSessionAttributes session) {
+
+        boolean isEmailNeeded = fsLogic.isFeedbackSessionForStudentsToAnswer(session);
```

Comment:
This means if the session is not for students, the closed email will not be sent. But is it a sufficient condition for the email not to be sent? A session might also contain questions only for instructors.